### PR TITLE
fix(@nestjs/swagger): Fix generated types for single string literals

### PR DESCRIPTION
**Pull request by gamedevsam**
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Single string literal types don't generate a type property metadata, leading to parsing errors.

Ex:

```ts
interface Test { stringLiteral: "one"  }
```

Results in:

```ts
{ stringLiteral: { required: true } }
```

Issue Number: N/A

## What is the new behavior?

Single string literals get assigned the type of String, ex:

```ts
{ stringLiteral: { required: true, type: () => String } }
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: lib/plugin/utils/plugin-utils.ts [hunk 1]
```diff
@@ -45,7 +46,7 @@ export function getTypeReferenceAsString(
   if (isNumber(type)) {
     return Number.name;
   }
-  if (isString(type)) {
+  if (isString(type) || isStringLiteral(type)) {
```
Comment (gamedevsam) [inline lib/plugin/utils/plugin-utils.ts:49]: This is the actual fix, the rest is formatting changes.

File: test/plugin/fixtures/string-literal.dto.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export const stringLiteralDtoText = `
+export class StringLiteralDto {
+  @ApiProperty()
+  valueOne: "one";
+  @ApiProperty()
+  valueTwo: "one" | "two";
```
Comment (gamedevsam) [inline test/plugin/fixtures/string-literal.dto.ts:6]: Union types continue to be typed as Object, so no breaking changes.

File: lib/plugin/utils/plugin-utils.ts (hunk 1 above)
Comment (kamilmysliwiec) [reply lib/plugin/utils/plugin-utils.ts:49]: If the rest is formatting changes, can you please revert them and only leave the actual fix + added tests?

Comment (gamedevsam) [reply lib/plugin/utils/plugin-utils.ts:49]: The formatting changes was the result of running `npm run format` as detailed in the `CONTRIBUTING.md` doc. I can revert, but it seems like this is the right thing to do before opening a PR?

**Comment by kamilmysliwiec** [comment]
LGTM. Merged to 8.0.0 branch

**Pull request merged**